msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLFileParser.java

```diff
@@ -62,12 +62,12 @@
  */
 public class MzMLFileParser implements MSDKMethod<RawDataFile> {
   private final @Nonnull File mzMLFile;
-  private final @Nonnull ArrayList<MsScan> spectrumList;
   private final ArrayList<MzMLReferenceableParamGroup> referenceableParamGroupList;
   private MzMLRawDataFile newRawFile;
-  private Integer lastScanNumber = 0;
   private volatile boolean canceled;
   private Float progress;
+  private int approxProgress;
```

Comment:
since this variable is only used for logger output, the name "approxProgress" is not very clear. how about something like "lastLoggedProgress"?